creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -772,6 +775,60 @@ class SnapbackSM {
       nodeUser.user_id % ModuloBase === this.currentModuloSlice
     )
   }
+
+  /**
+   * Issues syncRequest for user against secondary, and polls for replication up to primary
+   * If secondary fails to sync within specified timeoutMs, will error
+   */
+  async issueSyncRequestsUntilSynced (secondaryUrl, wallet, primaryClockVal, timeoutMs) {
+    // Issue syncRequest before polling secondary for replication
+    await this.enqueueSync({
+      userWallet: wallet,
+      secondaryEndpoint: secondaryUrl,
+      primaryEndpoint: this.endpoint,
+      syncType: SyncType.Manual,
+      immediate: true
+    })
+
+    // Poll clock status and issue syncRequests until secondary is caught up or until timeoutMs
+    const start = Date.now()
+    while (Date.now() - start < timeoutMs) {
+      try {
+        // Retrieve secondary clock status for user
+        const secondaryClockStatusResp = await axios({
+          method: 'get',
+          baseURL: secondaryUrl,
+          url: `/users/clock_status/${wallet}`,
+          responseType: 'json',
+          timeout: 1000 // 1000ms = 1s
+        })
+        const { clockValue: secondaryClockVal } = secondaryClockStatusResp.data.data
+
+        // If secondary is synced, return successfully
+        if (secondaryClockVal >= primaryClockVal) {
+          return
+        }
+
+        // Else, issue syncRequest to secondary
```

Comment:
does it make sense to keep re-enqueueing syncs here if a sync is already outstanding?

Reply:
it doesn't but as of now there is no way to know if there is one outstanding, i guess we could use sync status instead of clock status route, which would return 423 for in progress

Reply:
had an idea to address this as well, pushed up in latest commit lets discuss

Reply:
discussed offline, this implementation works